tests/test_permissions_synchronizer.py

```diff
@@ -20,82 +20,106 @@ class TestSyncPermissions(unittest.TestCase):
-            tmp.write(yaml.safe_dump(data))
-            tmp.seek(0)  # back to start since file still open (auto-delete if closed)
-            with contextlib.ExitStack() as stack:
-                stack.enter_context(mock.patch("cowbird.services.impl.magpie.Magpie",
-                                               side_effect=utils.MockMagpieService))
-                stack.enter_context(mock.patch("cowbird.services.impl.geoserver.Geoserver",
-                                               side_effect=utils.MockAnyService))
-                stack.enter_context(mock.patch("cowbird.services.impl.thredds.Thredds",
-                                               side_effect=utils.MockAnyService))
-                utils.get_test_app(settings={"cowbird.config_path": tmp.name})
-
-                resource_name = "resource1"
-                magpie = ServiceFactory().get_service("Magpie")
-
-                for svc in test_services:
-                    for perm_name in sync_perm_name[svc]:
-                        permission = Permission(
-                            service_name=svc,
-                            resource_id=0,
-                            resource_full_name=res_root[svc] + resource_name,
-                            name=perm_name,
-                            access="string1",
-                            scope="string2",
-                            user="string3")
-                        PermissionSynchronizer(magpie).create_permission(permission)
-                        assert len(magpie.json()["outbound_perms"]) == len(sync_perm_name[mapped_service[svc]])
-                        for idx, mapped_perm_name in enumerate(sync_perm_name[mapped_service[svc]]):
-                            outbound_perm = magpie.json()["outbound_perms"][idx]
-                            assert outbound_perm.service_name == mapped_service[svc]
-                            assert outbound_perm.resource_id == utils.MockAnyService.ResourceId
-                            assert outbound_perm.resource_full_name == res_root[mapped_service[svc]] + resource_name
-                            assert outbound_perm.name == mapped_perm_name
-                            assert outbound_perm.access == permission.access
-                            assert outbound_perm.scope == permission.scope
-                            assert outbound_perm.user == permission.user
-                        PermissionSynchronizer(magpie).delete_permission(permission)
-                        assert len(magpie.json()["outbound_perms"]) == 0
+        The PermissionSynchronizer create_permission function will first find if the applied permission exists in the
+        config. Then for every configured service it will obtain the  equivalent permission for this service and apply
+        it to the mocked Magpie service. The outbound_perm dict of the mocked Magpie service is then check to validate
+        that every permission that should have been created in Magpie exists.
```

Comment:
typos: 
- `sync config and check` -> `sync config and checks`
- space in `obtain  the equivalent`
- `is then check` -> `is then checked`

should used double ``` `` ```  for items that refer to code.